src/cinema/presentation/CinemaController.java

```diff
@@ -0,0 +1,61 @@
+package cinema.presentation;
+
+import cinema.businesslayer.CinemaService;
+import cinema.domainmodel.Seat;
+import cinema.domainmodel.Token;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+public class CinemaController {
+
+    @Autowired
+    CinemaService cinemaService;
+
+    @GetMapping("/seats")
+    public SeatsDto getSeats() {
+        return cinemaService.getSeats();
+    }
+
+    @PostMapping("/purchase")
+    public TicketDto purchaseTicket(@RequestBody Seat seat) {
+        if (!cinemaService.checkIfRowColumnValid(seat.getRow(), seat.getColumn())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The number of a row or a column is out of bounds!");
+        } else {
+            if (cinemaService.checkIfTicketPurchased(seat.getRow(), seat.getColumn())) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The ticket has been already purchased!");
+            } else {
+                return cinemaService.purchaseSeat(seat.getRow(), seat.getColumn());
+            }
+        }
+    }
+
+    @PostMapping("/return")
+    public ReturnedTicketDto returnTicket(@RequestBody Token token) {
+        if (!cinemaService.checkToken(token.getToken())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Wrong token!");
+        } else {
+            return cinemaService.returnTicket(token.getToken());
+        }
+    }
+
+    @PostMapping("/stats")
+    public StatisticDto getStatistic(@RequestParam("password") Optional<String> password) {
+        if (!password.equals(Optional.of("super_secret"))|| password.isEmpty()) {
```

Comment:
And here same for complexity here.

Also, it seems that if you check !"super_secret".equals(password) it would be enough